countComponents.js

```diff
@@ -0,0 +1,75 @@
+/*
+graph of nodes
+
+array[i] = [a, b]
+
+i: n of nodes
+o: => number of connected components
+
+Input: n = 5, edges = [[0,1],[1,2],[3,4]]
+Output: 2
+
+0 - 1 - 2
+
+3 - 4
+
+*/
+
+
+const buildGraph = (edges) => {
+    const graph = {};
+
+    for (const edge of edges) {
+        const [nodeOne, nodeTwo] = edge;
+
+        const isNodeOne = graph.hasOwnProperty(nodeOne);
+        const isNodeTwo = graph.hasOwnProperty(nodeTwo);
+
+        if (!isNodeOne) graph[nodeOne] = [];
+        if (!isNodeTwo) graph[nodeTwo] = [];
+
+        graph[nodeOne].push(nodeTwo);
+        graph[nodeTwo].push(nodeOne);
+    }
+
+    return graph;
+};
+
+const markAsVisited = (node, edges, visited) => {
+    const queue = new Queue();
+    queue.enqueue([node]);
+
+    const graph = buildGraph(edges);
+    while (queue.size() > 0) {
+        // Remove node
+        const [node] = queue.dequeue();
```

Comment:
It never really makes sense to destructure out a single value like this. If you only have a single value in a container (a container is something like an array or an object), then you might as well just remove the container and remove the need for destructuring.